src/software/sensor_fusion/possession/possession_tracker.cpp

```diff
@@ -36,7 +37,14 @@ TeamPossession PossessionTracker::getTeamWithPossession(const Team &friendly_tea
     else if ((time_near_friendly < time_near_threshold) &&
              (time_near_enemy > time_near_threshold))
     {
-        possession = TeamPossession::ENEMY_TEAM;
+        if (time_near_enemy < time_stagnant_threshold)
```

Comment:
I feel like stagnant position should also be tied to ball position. It'd be more appropriate to transition to aggressive, ball-stealing defence if the ball hasn't moved considerably between subsequent ticks

Reply:
Added a check for ball stagnancy